### PR TITLE
tlsn-core fixtures

### PR DESCRIPTION
**Pull request by sinui0**
This PR moves the fixtures (again) into `tlsn-core` and eliminates a cyclic dependency that currently exists between `tlsn-core` and `tlsn-fixtures`.

I would prefer to have the fixtures in a separate crate but we need to use them for tests in `tlsn-core` so this is path of least resistance without a bunch of duplication.

# Changes
- Moved `tlsn-fixtures` into a module in `tlsn-core`
- Added a feature flag `fixtures` to `tlsn-core`
- Added running tests with `--all-features` in our CI
- Added a suite of functions for constructing fixtures.

### CODE REVIEW

File: tlsn/tlsn-core/src/fixtures/cert.rs [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+    #[fixture]
+    fn cert_verifier() -> impl ServerCertVerifier {
+        super::cert_verifier()
+    }
+
+    /// Expect chain verification to succeed
+    #[rstest]
+    #[case(tlsnotary())]
+    #[case(appliedzkp())]
+    fn test_verify_cert_chain_sucess_ca_implicit(
```
Comment (themighty1) [inline tlsn/tlsn-core/src/fixtures/cert.rs:165]:
```suggestion
    fn test_verify_cert_chain_success_ca_implicit(
```

File: tlsn/tlsn-core/src/fixtures/cert.rs [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+                &mut std::iter::empty(),
+                &[],
+                SystemTime::UNIX_EPOCH + Duration::from_secs(data.time),
+            )
+            .is_err());
+    }
+
+    /// Expect to fail because the host name is not a valid DNS name
+    #[rstest]
+    fn test_check_dns_name_present_in_cert_fail_invalid_dns_name() {
```
Comment (themighty1) [inline tlsn/tlsn-core/src/fixtures/cert.rs:366]: seems like this test should go to components/tls/tls-core/src/dns.rs

**Review by themighty1 — APPROVED** [review]
gw, some minor comments

**Review by th4s — APPROVED** [review]
:+1:

File: tlsn/tlsn-core/src/fixtures/cert.rs (hunk 2 above)
Comment (themighty1) [reply tlsn/tlsn-core/src/fixtures/cert.rs]: ok

**Pull request merged**